internal/performance/export_test.go

```diff
@@ -0,0 +1,259 @@
+		criteria = publishdb.IterateExposuresCriteria{
+			OnlyLocalProvenance: false,
+		}
+		want           = keysPerPublish * numPublishes
+		batchStartTime = time.Now().Add(time.Duration(-totalBatches-10) * exportPeriod)
+	)
+
+	env, client, db := integration.NewTestServer(t, exportPeriod)
+	payload := &verifyapi.Publish{
+		Keys:           util.GenerateExposureKeys(keysPerPublish, -1, false),
+		Regions:        []string{"TEST"},
+		AppPackageName: "com.example.app",
+
+		// TODO: hook up verification
+		VerificationPayload: "TODO: ",
+	}
+	if err := client.PublishKeys(payload); err != nil {
+		t.Fatal(err)
+	}
+
+	var exposures []*publishmodel.Exposure
+	if _, err := publishdb.New(db).IterateExposures(ctx, criteria, func(m *publishmodel.Exposure) error {
+		exposures = append(exposures, m)
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+	if l := len(exposures); l != keysPerPublish {
+		t.Fatalf("Want: %d keys, got: %d", keysPerPublish, l)
+	}
+
+	// Creates batche based on first batch
+	time.Sleep(3 * time.Second)
+	if err := client.ExportBatches(); err != nil {
+		t.Fatal(err)
+	}
+	if err := db.InTx(ctx, pgx.Serializable, func(tx pgx.Tx) error {
+		result, err := tx.Exec(ctx, `
+					UPDATE
+						ExportBatch
+					SET
+						start_timestamp = $1,
+						end_timestamp = $2
+					WHERE
+						start_timestamp > $3
+				`,
+			batchStartTime,
+			batchStartTime.Add(exportPeriod),
+			time.Now().Add(-2*exportPeriod),
+		)
+		if err != nil {
+			return err
+		}
+		if got, want := result.RowsAffected(), int64(1); got != want {
+			return fmt.Errorf("expected %v to be %v", got, want)
+		}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+	// Delete the first batch of exposures. The creation time of all exposures
+	// in this test are modified to fit nicely among exports, delete the first
+	// one is easier than running a sql modifying them
+	deleted, err := publishdb.New(db).DeleteExposuresBefore(ctx, time.Now())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if deleted != keysPerPublish {
+		t.Fatalf("Delete template publishes, want: %d, got: %d", keysPerPublish, deleted)
+	}
+
+	// Publish keys based on first batch of published keys
+	for i := 0; i < numPublishes; i++ {
+		if r := i % roughPerBatch; r == 0 { // increace start time after each batch
+			batchStartTime = batchStartTime.Add(exportPeriod)
+		}
+		var revisedExposures []*publishmodel.Exposure
+		for j, newKey := range util.GenerateExposureKeys(keysPerPublish, -1, false) {
+			m := *exposures[j]
+			m.CreatedAt = batchStartTime.Add(1 * time.Second)
+			m.ExposureKey, _ = base64util.DecodeString(newKey.Key)
+			revisedExposures = append(revisedExposures, &m)
+		}
+		updated, err := publishdb.New(db).InsertAndReviseExposures(ctx, revisedExposures)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if updated != keysPerPublish {
+			t.Fatalf("Want updated: %d, got %d", keysPerPublish, updated)
+		}
+	}
+
+	// Ensure all keys published
+	integration.Eventually(t, 30, func() error {
```

Comment:
is this step needed? doesn't finding all the keys in and export tell us that?

Reply:
Good point, it's not needed. Deleted